### ISSUE TITLE
inconsistent back button in app bar when using child routes

### ISSUE THREAD

**Issue by asequeira-os** [Milad-Akarie/auto_route_library#1926]
Trying to use child routes.
Created a sample app to try out.
Please see https://github.com/asequeira-os/routenoback

In that app, pressing the buttons `inner A 1` or `inner A2` causes the display of the inner route widgets.
But it does not show back arrow in the app bar.

Now if I press `outerr a` then follow up with pressing `inner A 1` or 2, I do see a back button.

Overall I am a bit confused on the expected behavior as to if this is my misunderstanding, or my example bug or bug in auto route.

Basically I'd like the ability to have child routes that can be pushed.

Please help.

**Comment by asequeira-os** [comment]
I think this is me mixing up navigator scopes.
